Lib/fontTools/subset/svg.py

```diff
@@ -0,0 +1,235 @@
+from __future__ import annotations
+
+import re
+from itertools import chain, count, groupby
+from typing import Dict, Iterable, Iterator, List, Set, Tuple
+
+try:
+    from lxml import etree
+except ModuleNotFoundError:
+    # lxml is required for subsetting SVG, but we prefer to delay the import error
+    # until subset_glyphs() is called (i.e. if font to subset has an 'SVG ' table)
+    etree = None
+
+from fontTools import ttLib
+from fontTools.subset.util import _add_method
+
+
+__all__ = ["subset_glyphs"]
+
+
+GID_RE = re.compile(r"^glyph(\d+)$")
+
+NAMESPACES = {
+    "svg": "http://www.w3.org/2000/svg",
+    "xlink": "http://www.w3.org/1999/xlink",
+}
+XLINK_HREF = f'{{{NAMESPACES["xlink"]}}}href'
+
+if etree is not None:
+    # it's more efficient to compile XPath objects upfront than calling the same
+    # Element.xpath() several times
+    find_svg_elements_with_id = etree.XPath(".//svg:*[@id]", namespaces=NAMESPACES)
```

Comment:
Optional, another approach would be to return these from a function with a functools annotation (lru_cache or cache depending on what the minimum python fonttools allows is).

Spares the etree None check because those functions are internal and only called from inside functions that have done the throw if no etree step.

Reply:
thanks but I think using functools.lru_cache is a bit overkill for this

Reply:
Is it heavy? - I think I'd have thought of that as "lighter" than running code every time this file loads but maybe my perception of python "weight" is off

Reply:
alright, I added lru_cache https://github.com/fonttools/fonttools/pull/2452/commits/f1e9241e9f7ed94247471efa1429d83ee31bd9cd